### PR TITLE
query testing & split for sorting

### PR DESCRIPTION
**Pull request by n1mus**
- [ ] I updated the README.md docs to reflect this change.

For changes to the codebase:

- [x] I have written tests to cover this change.
- [x] This is not a breaking API change OR
- [ ] This is a breaking API change and I have incremented the API version and added a summary to CHANGELOG.md.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,3 +18,6 @@ coverage_report/
 
 # Spec archives
 spec.tar.gz
+
+# Creds
+arango_live_server_config.json
```
Comment (n1mus) [inline .gitignore:23]: This should have all the info to log in using the python ArangoDB client. Username, password, url

File: scripts/run_tests.sh [hunk 2]
```diff
@@ -17,7 +17,7 @@ coverage erase
 # spec validation
 python -m spec.validate
 # run importer/, relation_engine_server/, and spec/ tests
-coverage run --branch -m unittest discover -v
+coverage run --branch -m pytest
```
Comment (n1mus) [inline scripts/run_tests.sh:20]: unittest works well enough but pytest tries harder!

File: spec/stored_queries/taxonomy/taxonomy_ncbi_species.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
+# Search ncbi_taxon collection for species/strains by scientific name
```
Comment (n1mus) [inline spec/stored_queries/taxonomy/taxonomy_ncbi_species.yaml:1]: New ncbi species/strain search

File: spec/stored_queries/taxonomy/taxonomy_ncbi_species_no_sort.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
+# Search ncbi_taxon collection for species/strains by scientific name
```
Comment (n1mus) [inline spec/stored_queries/taxonomy/taxonomy_ncbi_species_no_sort.yaml:1]: New ncbi species/strain search, but with no sorting, for when the length of search text is low

**Comment by n1mus** [comment]
IDK if I should add my Google docs "ADR" as a pdf somewhere under an `adr` directory

File: scripts/run_tests.sh (hunk 2 above)
Comment (ialarmedalien) [reply scripts/run_tests.sh:20]: tries harder?

File: dev-requirements.txt [hunk 5]
```diff
@@ -7,3 +7,5 @@ coverage==5.2.1
 typed-ast>=1.4.0
 black==20.8b1
 pytest==6.2.5
+python-arango==5.4.0
+numpy==1.21.2
```
Comment (ialarmedalien) [inline dev-requirements.txt:11]: Things are getting serious now!

File: spec/test/stored_queries/test_query.py [hunk 6]
```diff
@@ -0,0 +1,436 @@
...
+Please set host URL, username, and password in arango_live_server_config.json, e.g.,
+{
+    "username": "doe_j",
+    "password": "cat-sat-hat",
+    "host": "http://10.58.1.211:8532"
+}
+Note: if you are on a local machine
+you may have to proxy into the live ArangoDB server first, e.g.,
+`ssh -L 8532:10.58.1.211:8532 j_doe@login1.berkeley.kbase.us`
+Then, the url would be `http://localhost:8532`
```
Comment (ialarmedalien) [inline spec/test/stored_queries/test_query.py:57]: nice! we need more useful documentation like this 👍

File: spec/test/stored_queries/test_query.py [hunk 7]
```diff
@@ -0,0 +1,436 @@
...
+    Somewhat arbitrary determination of whether a fulltext's search text is "simple"
+    relative to its search time
+    """
+    return len(search_text.split()) == 2 and all(
+        [tok.isalnum() and len(tok) >= 3 for tok in search_text.split()]
+    )
+
+
+def jprint(jo, dry=False):
+    txt = json.dumps(jo, indent=3)
```
Comment (ialarmedalien) [inline spec/test/stored_queries/test_query.py:107]: I also like having `sort_keys` turned on to make it easier to find whatever you're looking for

File: spec/test/stored_queries/test_query.py [hunk 8]
```diff
@@ -0,0 +1,436 @@
...
+    resample=True,
+    cap_scinames=2000,
+    cap_scinames_prefixes=5000,
+):
+    """
+    Get samplings of scinames or prefixes thereof to gauge execution time
+
+    Things to include:
+    * Simple genus/species epithets with two non-short words
+    * "Wild" scientific names, defined as the exclusion of the simple scientific names
```
Comment (ialarmedalien) [inline spec/test/stored_queries/test_query.py:142]: I think that the "simple scientific names" are usually called "binomial" identifiers, so you could call the wild ones "non-binomial" or something like "complex" or "non-canonical"

**Review by ialarmedalien — CHANGES_REQUESTED** [review]
Some comments and one minor change to use your `jprint` function.

File: scripts/run_tests.sh (hunk 2 above)
Comment (ialarmedalien) [reply scripts/run_tests.sh:20]: and colour!

File: spec/test/stored_queries/test_query.py (hunk 8 above)
Comment (n1mus) [reply spec/test/stored_queries/test_query.py:142]: yeah ... didn't think this would end up in the repo hehe

**Pull request merged**